service/mapper/pathings_test.go

```diff
@@ -0,0 +1,147 @@
+	testPath3 = ledger.Path{0x60, 0x0a, 0xd8, 0xa4, 0xf1, 0x6b, 0xce, 0x2e, 0x57, 0x59, 0xfd, 0x6e, 0x45, 0xcf, 0xa9, 0xa0, 0x60, 0x0a, 0xd8, 0xa4, 0xf1, 0x6b, 0xce, 0x2e, 0x57, 0x59, 0xfd, 0x6e, 0x45, 0xcf, 0xa9, 0xa0}
+	testPath4 = ledger.Path{0xa5, 0x68, 0x7b, 0x2d, 0x95, 0x18, 0x7b, 0xc7, 0xce, 0xd0, 0xe1, 0x02, 0xd6, 0xce, 0xfe, 0x93, 0xa5, 0x68, 0x7b, 0x2d, 0x95, 0x18, 0x7b, 0xc7, 0xce, 0xd0, 0xe1, 0x02, 0xd6, 0xce, 0xfe, 0x93}
+	testPath5 = ledger.Path{0x60, 0x0a, 0xd8, 0xa4, 0xf1, 0x6b, 0xce, 0x2e, 0x57, 0x59, 0xfd, 0x6e, 0x45, 0xcf, 0xa9, 0xa0, 0x60, 0x0a, 0xd8, 0xa4, 0xf1, 0x6b, 0xce, 0x2e, 0x57, 0x59, 0xfd, 0x6e, 0x45, 0xcf, 0xa9, 0xa0}
+	testPath6 = ledger.Path{0xfd, 0x84, 0xc0, 0xa7, 0xb2, 0x35, 0xc9, 0x89, 0xc1, 0x8e, 0x6a, 0xa2, 0x69, 0x04, 0xfe, 0xba, 0xfd, 0x84, 0xc0, 0xa7, 0xb2, 0x35, 0xc9, 0x89, 0xc1, 0x8e, 0x6a, 0xa2, 0x69, 0x04, 0xfe, 0xba}
+	testKey   = ledger.NewKey([]ledger.KeyPart{
+		ledger.NewKeyPart(0, []byte(`owner`)),
+		ledger.NewKeyPart(1, []byte(`controller`)),
+		ledger.NewKeyPart(2, []byte(`key`)),
+	})
+	testValue1   = ledger.Value(`test1`)
+	testValue2   = ledger.Value(`test2`)
+	testValue3   = ledger.Value(`test3`)
+	testValue4   = ledger.Value(`test4`)
+	testValue5   = ledger.Value(`test5`)
+	testValue6   = ledger.Value(`test6`)
+	testPayload1 = ledger.NewPayload(testKey, testValue1)
+	testPayload2 = ledger.NewPayload(testKey, testValue2)
+	testPayload3 = ledger.NewPayload(testKey, testValue3)
+	testPayload4 = ledger.NewPayload(testKey, testValue4)
+	testPayload5 = ledger.NewPayload(testKey, testValue5)
+	testPayload6 = ledger.NewPayload(testKey, testValue6)
+
+	testPaths    = []ledger.Path{testPath1, testPath2, testPath2, testPath2, testPath3, testPath4}
+	testPayloads = []*ledger.Payload{testPayload1, testPayload2, testPayload3, testPayload4, testPayload5, testPayload6}
+
+	testUpdate = &ledger.TrieUpdate{
+		Paths:    testPaths,
+		Payloads: testPayloads,
+	}
+
+	// Test nodes visual representation:
+	//           6 (root)
+	//          / \
+	//         3   5
+	//        / \   \
+	//       1   2   4
+	//
+	testNode1 = node.NewLeaf(testPath1, testPayload1, 256)
+	testNode2 = node.NewLeaf(testPath2, testPayload2, 256)
+	testNode3 = node.NewNode(256, testNode1, testNode2, testPath3, testPayload3, hash.DummyHash, 64, 64)
+	testNode4 = node.NewLeaf(testPath4, testPayload4, 256)
+	testNode5 = node.NewNode(256, testNode4, nil, testPath5, testPayload5, hash.DummyHash, 64, 64)
+	testRoot  = node.NewNode(256, testNode3, testNode5, testPath6, testPayload6, hash.DummyHash, 64, 64)
+)
+
+func TestAllPaths(t *testing.T) {
+	t.Run("nominal case with single path", func(t *testing.T) {
+		t.Parallel()
+
+		testPath := ledger.Path{0xaa, 0xc5, 0x13, 0xeb, 0x1a, 0x04, 0x57, 0x70, 0x0a, 0xc3, 0xfa, 0x8d, 0x29, 0x25, 0x13, 0xe1}
```

Comment:
This is the dangerous part I just talked about on Slack. There are two things that can happen here:

1) The type is an array, in which case missing values are filled with zeros, potentially leading to unexpected values when those zeros are included in other places.
2) The type is a slice, in which case it does not have the correct length that a path should have, potentially leading to things breaking unexpectedly in certain places.

That's why we should only use `ledger.Path` with all bytes set (32) and enforce this in all functions were a path does not come directly from Flow (i.e. DPS API, Rosetta API, or our own index). IMO only paths coming from Flow should be assumed to be correct, and even then, one could argue for defensive programming.